diff --git a/arch/arm/boot/dts/microchip/pm9g45.dts b/arch/arm/boot/dts/microchip/pm9g45.dts
index c349fd375..d46c742ab 100644
--- a/arch/arm/boot/dts/microchip/pm9g45.dts
+++ b/arch/arm/boot/dts/microchip/pm9g45.dts
@@ -110,29 +110,34 @@ at91bootstrap@0 {
 							reg = <0x0 0x20000>;
 						};
 
-						barebox@20000 {
-							label = "barebox";
-							reg = <0x20000 0x40000>;
+						uboot@20000 {
+							label = "u-boot";
+							reg = <0x20000 0xe0000>;
 						};
 
-						bareboxenv@60000 {
-							label = "bareboxenv";
-							reg = <0x60000 0x1A0000>;
+						ubootenv@100000 {
+							label = "uboot env";
+							reg = <0x100000 0x40000>;
 						};
 
-						kernel@200000 {
-							label = "bareboxenv2";
-							reg = <0x200000 0x300000>;
+						ubootenv2@140000 {
+							label = "uboot env";
+							reg = <0x140000 0x40000>;
+						};
+
+						dtb@180000 {
+							label = "device tree";
+							reg = <0x180000 0x800000>;
 						};
 
-						kernel@500000 {
-							label = "root";
-							reg = <0x500000 0x400000>;
+						kernel@200000 {
+							label = "kernel";
+							reg = <0x200000 0x600000>;
 						};
 
-						data@900000 {
-							label = "data";
-							reg = <0x900000 0x8340000>;
+						data@800000 {
+							label = "rootfs";
+							reg = <0x800000 0x8000000>;
 						};
 					};
 				};
